### PR TITLE
Feature/turret shooting

### PR DESCRIPTION
**Pull request by YiHChen**
- Turrets now have an inventory
- Inventory is given a unique 'turretGun' upon creation 
- turretGun currently only has 10 shots (not sure about these stats at the moment, will most likely change)
-Upon reaching 0 ammo, turret is automatically destroyed.

### CODE REVIEW

File: src/game/GameManager.cpp [hunk 1]
```diff
@@ -409,7 +446,7 @@ WeaponDrop& GameManager::getWeaponDrop(const int32_t id) {
 //returns weapon in weaponManager using id
 std::shared_ptr<Weapon> GameManager::getWeapon(const int32_t id) {
     const auto& w = weaponManager[id];
-    assert(w.second);
+    //assert(w.second);
```
Comment (JohnAgapeyev) [inline src/game/GameManager.cpp]: No,
A thousand times no
Why did you remove a perfectly good assert?
No

File: src/inventory/weapons/TurretGun.h [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+    constexpr int DAMAGE        = 75;
+    constexpr int AOE           = 1;
+    constexpr int PENETRATION   = 3;
+
+    constexpr int CLIP          = 10;
+    constexpr int CLIPMAX       = 10;
+    constexpr int AMMO          = 10;
+
+    constexpr int RELOADDELAY   = 100;
+    constexpr int FIREDELAY     = 1500;
```
Comment (JohnAgapeyev) [inline src/inventory/weapons/TurretGun.h]: This is  some nasty spacing

File: src/turrets/Turret.cpp [hunk 3]
```diff
@@ -166,20 +160,10 @@ void Turret::collidingProjectile(const int damage) {
...
- * Function Interface: bool ammoCheckTurret()
- * Description:
- * Checks the current amount of ammo for the turret's gun. If the current amount of
- * ammo is greater than 0, the function returns true and false otherwise.
- */
-bool Turret::ammoCheckTurret() {
-    // This is going to be changed soon so it is no longer a simple getter.
-    return (ammo > 0);
+    Weapon *w = inventory.getCurrent();
+    if (w != nullptr) {
```
Comment (JohnAgapeyev) [inline src/turrets/Turret.cpp]: `if (w) {`

File: src/turrets/Turret.h [hunk 4]
```diff
@@ -61,18 +60,22 @@ class Turret : public Movable {
...
     bool isPlaced() {return placed;};
     void pickUpTurret();
+    bool isActivated() {return activated;};
     void removeTurret(); // removes the turret
     // Jamie Lee, 2017-03-01.
     float getRange() const {
         return range;
     }
 
+    Inventory inventory;
```
Comment (JohnAgapeyev) [inline src/turrets/Turret.h]: You made a public member
John no likey public members
Fix this

**Review by JohnAgapeyev — CHANGES_REQUESTED** [review]

**Review by JohnAgapeyev — APPROVED** [review]

File: src/inventory/weapons/TurretGun.h [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+#define TURRETGUN_H
+#include <string>
+
+#include "InstantWeapon.h"
+#include "../../audio/AudioManager.h"
+#include "../../sprites/Renderer.h"
+#include "../../sprites/SpriteTypes.h"
+
+
+using std::string;
```
Comment (isaacmorneau) [inline src/inventory/weapons/TurretGun.h:14]: while this is valid and should be ok, avoid all usings in headers. it is kinder to people including the files to not have namespace lookups they dont want.

File: src/turrets/Turret.cpp [hunk 6]
```diff
@@ -1,7 +1,11 @@
 /**
- * Source: Turret.cpp  
- *
- * Functions: void placementCheckTurret()
+ * Date: Feb. 02, 2017
+ * Designer: Mark Chen, Jaime Lee, Terry Kang
+ * Programmer: Mark Chen, Micheal Goll, Jacob McPhail, Isaac Morneau, Maitiu Morton, Mark Tattrie
```
Comment (isaacmorneau) [inline src/turrets/Turret.cpp]: um, really? did we all work on this?

File: src/turrets/Turret.h [hunk 7]
```diff
@@ -61,18 +60,24 @@ class Turret : public Movable {
     void placeTurret();
     bool isPlaced() {return placed;};
     void pickUpTurret();
+    bool isActivated() {return activated;};
     void removeTurret(); // removes the turret
+    Inventory getInventory() {return inventory;};
```
Comment (isaacmorneau) [inline src/turrets/Turret.h:65]: should be a reference

**Review by isaacmorneau — CHANGES_REQUESTED** [review]

**Review by isaacmorneau — APPROVED** [review]

**Comment by isaacmorneau** [comment]
merge conflicts

**Pull request merged**